src/meta/ClientBasedGflagsManager.cpp

```diff
@@ -0,0 +1,142 @@
+}
+
+folly::Future<StatusOr<bool>>
+ClientBasedGflagsManager::setConfig(const cpp2::ConfigModule& module, const std::string& name,
+                                    const cpp2::ConfigType& type, const VariantType &value) {
+    switch (type) {
+        case cpp2::ConfigType::INT64:
+            return set(module, name, type, boost::get<int64_t>(value));
+        case cpp2::ConfigType::DOUBLE:
+            return set(module, name, type, boost::get<double>(value));
+        case cpp2::ConfigType::BOOL:
+            return set(module, name, type, boost::get<bool>(value));
+        case cpp2::ConfigType::STRING:
+            return set(module, name, type, boost::get<std::string>(value));
+        default:
+            return Status::Error("parse value type error");
+    }
+}
+
+folly::Future<StatusOr<std::vector<cpp2::ConfigItem>>>
+ClientBasedGflagsManager::getConfig(const cpp2::ConfigModule& module, const std::string& name) {
+    return metaClient_->getConfig(module, name);
+}
+
+folly::Future<StatusOr<std::vector<cpp2::ConfigItem>>>
+ClientBasedGflagsManager::listConfigs(const cpp2::ConfigModule& module) {
+    return metaClient_->listConfigs(module);
+}
+
+folly::Future<StatusOr<bool>>
+ClientBasedGflagsManager::registerConfig(const cpp2::ConfigModule& module, const std::string& name,
+                                         const cpp2::ConfigType& type, const cpp2::ConfigMode& mode,
+                                         const std::string& value) {
+    auto item = toThriftConfigItem(module, name, type, mode, value);
+    std::vector<cpp2::ConfigItem> items;
+    items.emplace_back(item);
+    return metaClient_->regConfig(items);
+}
+
+void ClientBasedGflagsManager::loadCfgThreadFunc() {
+    // only load current module's config is enough
+    auto ret = listConfigs(module_).get();
+    if (ret.ok()) {
+        auto items = ret.value();
+        updateConfigCache(items);
+    } else {
+        LOG(INFO) << "Load configs failed: " << ret.status();
+    }
+
+    std::string name = "load_config_interval_secs";
+    gflags::CommandLineFlagInfo interval;
+    gflags::GetCommandLineFlagInfo(name.c_str(), &interval);
+    size_t delayMS = folly::to<int64_t>(interval.default_value) * 1000;
+    try {
+        delayMS = folly::to<int64_t>(interval.current_value) * 1000;
+    } catch (const std::exception& ex) {
+        LOG(ERROR) << "Parse config " << name << "error: " << ex.what();
+    }
+    bgThread_.addDelayTask(delayMS + folly::Random::rand32(900),
+                           &ClientBasedGflagsManager::loadCfgThreadFunc, this);
+    LOG(INFO) << "Load configs after " << delayMS << " ms";
+}
+
+void ClientBasedGflagsManager::regCfgThreadFunc() {
+    auto status = metaClient_->regConfig(gflagsDeclared_).get();
+    if (status.ok()) {
+        LOG(INFO) << "Register flags ok " << gflagsDeclared_.size();
+        return;
+    }
+
+    std::string name = "load_config_interval_secs";
```

Comment:
Why the snip code about "load_config_interval_secs" flag

Reply:
I will modify the init process.